pkg/dns/service.go

```diff
@@ -159,14 +174,16 @@ func (s *Service) SetEndpoints(ctx context.Context, addresses []gatewayv1beta1.G
 	old := dnsRecord.DeepCopy()
 	host := dnsRecord.Name
 
-	// check if endpoint already exists in the DNSRecord
+	// check if endpoint already exists in the DNSRecord - we cannot delete and rebuild as health points are injected into
+	// these endpoints elsewhere - TODO refactor this to be all in one place.
```

Comment:
do you want to make sure there is an issue for this?

Reply:
I'd guess this might fall under the work for health checks.

Reply:
https://github.com/Kuadrant/multicluster-gateway-controller/issues/294